lib/Mojo/UserAgent/Server.pm

```diff
@@ -12,11 +12,16 @@ sub app {
 
   # Singleton application
   state $singleton;
-  return $singleton = $app ? $app : $singleton unless ref $self;
+  unless (ref $self) {
+    my $starting = ref $app && (!defined $singleton || $singleton ne $app);
+    $app->worker_starting if $starting;
+    return $singleton = $app ? $app : $singleton;
+  }
```

Comment:
I do not want to see any changes in this module.

Reply:
OK, fixed.

Reply:
To be more precise, i want it to work for `Mojo::UserAgent::Server`, but without any changes to the actual module. The feature needs to stay contained to `Mojo::Server::Daemon`.

Reply:
If there's no elegant solution, then `Mojo::IOLoo->next_tick` is simply better in my opinion.

Reply:
Done, there is no need to change UserAgent::Server after all.